### PR TITLE
Replace unnecessary `&mut self` with `&self` in `BlockImport::import_block()`

### PR DESCRIPTION
**Pull request by nazar-pc**
There was no need for it to be `&mut self` since block import can happen concurrently for different blocks and in many cases it was `&mut Arc<dyn BlockImport>` anyway :man_shrugging:

Similar in nature to https://github.com/paritytech/polkadot-sdk/pull/4844

### CODE REVIEW

**Comment by nazar-pc** [comment]
Is there some kind of tooling for semver? There are a lot of files changed and I'd hate to go through each crate manually.

**Comment by ggwpez** [comment]
> Is there some kind of tooling for semver? There are a lot of files changed and I'd hate to go through each crate manually.

Yea am gonna try to run the bot now for this. Just added it now https://github.com/paritytech/polkadot-sdk/pull/5331

**Comment by ggwpez** [comment]
Looks like the bot failed. Ran the script manually now. You probably want to demote some of these `majors` to `patch` if they are not publicly exposed. This prdoc is necessary as to not break SemVer for parachains.

<details><summary>Details</summary>
<p>

```pre
crates:
- bump: major
  name: cumulus-client-consensus-common
- bump: major
  name: cumulus-client-network
- bump: major
  name: cumulus-relay-chain-inprocess-interface
- bump: major
  name: cumulus-pallet-parachain-system
- bump: major
  name: sc-basic-authorship
- bump: major
  name: sc-consensus-babe
- bump: major
  name: sc-consensus-beefy
- bump: major
  name: sc-consensus
- bump: major
  name: sc-consensus-grandpa
- bump: major
  name: sc-consensus-pow
- bump: major
  name: mmr-gadget
- bump: major
  name: sc-network
- bump: major
  name: sc-network-sync
- bump: major
  name: sc-offchain
- bump: major
  name: sc-rpc-spec-v2
- bump: major
  name: sc-rpc
- bump: major
  name: sc-service
- bump: major
  name: sc-transaction-pool
doc:
- audience: Node Dev
  description: "There was no need for it to be `&mut self` since block import can\
    \ happen concurrently for different blocks and in many cases it was `&mut Arc<dyn\
    \ BlockImport>` anyway :man_shrugging:\r\n\r\nSimilar in nature to https://github.com/paritytech/polkadot-sdk/pull/4844"
title: "Replace unnecessary `&mut self` with `&self` in `BlockImport::import_block()`"
```

</p>
</details>

**Comment by nazar-pc** [comment]
Nice, thank you!

File: substrate/test-utils/client/src/client_ext.rs [hunk 1]
```diff
@@ -154,8 +151,9 @@ where
 	RA: Send,
```
Comment (bkchr) [inline substrate/test-utils/client/src/client_ext.rs:151]:
```suggestion
	RA: Send + Sync,
```

File: substrate/test-utils/client/src/client_ext.rs [hunk 2]
```diff
@@ -154,8 +151,9 @@ where
...
 	B: Send + Sync,
 	E: Send + Sync,
+	RA: Send + Sync,
```
Comment (bkchr) [inline substrate/test-utils/client/src/client_ext.rs:154]:
```suggestion
```

**Review by bkchr — APPROVED** [review]

**Review by skunert — APPROVED** [review]

**Review by davxy — APPROVED** [review]

**Pull request merged**